Source/CBL_Router.m

```diff
@@ -553,12 +553,12 @@ - (void) processRequestRanges {
     NSUInteger from, to;
     if (fromStr.length > 0) {
         from = (NSUInteger)fromStr.integerValue;
-        if (toStr.length > 0)
+        if (toStr.length > 0) {
             to = MIN((NSUInteger)toStr.integerValue, bodyLength - 1);
-        else
+            if (to < from)
+                return;  // invalid range
+        } else
```

Comment:
This would incorrectly make "200-201" an invalid range if the bodyLength is 100... because `from` would be 200 and `to` would be 99.
I think you need to do the `MIN` operation after the `to < from` comparison.